<launch>

<arg name="odom_topic" default="/odom"/>
<arg name="init_x" value="2.0" />
<arg name="init_y" value="1.0" />
<arg name="init_z" value="1.0" />


<!-- planner -->
	<node pkg="planner_manager" name="scf_planner_node" type="scf_planner_node" output="screen">

	  <param name="occupancy_resolution" value="0.2"/>
	  <param name="sta_threshold"   value="1"/>
	  <param name="debug_output"   value="false"/>
	  
	  <!-- the distance between each neighboring waypoint of MINCO traj initalization -->
	  <param name="traj_parlength"   value="1.5"/>		
	  
	  <!-- number of samples in each traj segment  -->
	  <param name="optimization/K"     value="8"  		/>
	  <!-- esdf turcation distance -->
	  <param name="optimization/pok"   value="1.0"		/>
	  <!-- UAV Dynamic -->
	  <param name="optimization/vmax"  value="3.0" 		/>
	  <param name="optimization/amax"  value="5.0"		/>
	  <!-- weight -->
	  <param name="optimization/rhoT"  value="1000.0"	/>
	  <param name="optimization/rhoP"  value="10000.0"/>
	  <param name="optimization/rhoV"  value="1000.0"	/>
	  <param name="optimization/rhoA"  value="1000.0"	/>
	  
	  <remap from="~globalmap"     			to="/globalmap" />
	  <remap from="~globalmap_vis"			to="/globalmap_vis" />
	  <remap from="~gridmap_vis" 			to="/gridmap_vis" />
	  <remap from="~voxel_vis"			 	to="/voxel_vis" />
	  <remap from="~rcvmap_signal"			to="/rcvmap_signal" />
	  <remap from="~odom"					to="$(arg odom_topic)"/>


	  <remap from="~trajectory"			to="/trajectory" />
	  
	</node>

<!-- trajectory server -->
	<node pkg="planner_manager" name="traj_server" type="traj_server" output="screen">

	  <remap from="~trajectory_topic"     	to="/trajectory" />
	  <remap from="~odom" 					to="$(arg odom_topic)" />
	  <remap from="~point/vis"  			to="/vis_p" />
	  <remap from="~controller_cmd"  		to="/planning/pos_cmd" />

	</node>

<!-- use simulator -->
	<include file="$(find planner_manager)/launch/simulator.xml">
		<arg name="odometry_topic" value="$(arg odom_topic)" />
		<arg name="init_x" value="$(arg init_x)" />
		<arg name="init_y" value="$(arg init_y)" />
		<arg name="init_z" value="$(arg init_z)" />
	</include>

<!-- use program code globalmap generator -->
	<include file="$(find globalmap_gene)/launch/globalmap_gene.launch">
		<arg name="map_id" value="1" />
		<arg name="init_x" value="$(arg init_x)" />
		<arg name="init_y" value="$(arg init_y)" />
		<arg name="init_z" value="$(arg init_z)" />
	</include>


<!-- rviz -->
    	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find map_manager)/param/default.rviz" required="true" />

</launch>
